huaweicloud/services/ces/resource_huaweicloud_ces_dashboard.go

```diff
@@ -0,0 +1,298 @@
+				Optional:    true,
+				Computed:    true,
+				Description: `Specifies the enterprise project ID of the dashboard.`,
+			},
+			"dashboard_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `Specifies the copied dashboard ID.`,
+			},
+			"is_favorite": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: `Specifies whether the dashboard is favorite.`,
+			},
+			"creator_name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The creator name of the dashboard.`,
+			},
+			"created_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The creation time of the dashboard.`,
+			},
+		},
+	}
+}
+
+func resourceDashboardCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var (
+		httpUrl = "v2/{project_id}/dashboards"
+		product = "ces"
+	)
+	client, err := cfg.NewServiceClient(product, region)
+	if err != nil {
+		return diag.Errorf("error creating CES client: %s", err)
+	}
+
+	path := client.Endpoint + httpUrl
+	path = strings.ReplaceAll(path, "{project_id}", client.ProjectID)
+
+	opt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+	opt.JSONBody = utils.RemoveNil(buildCreateDashboardBodyParams(d, cfg))
+	resp, err := client.Request("POST", path, &opt)
+	if err != nil {
+		return diag.Errorf("error creating CES dashboard: %s", err)
+	}
+
+	respBody, err := utils.FlattenResponse(resp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id, err := jmespath.Search("dashboard_id", respBody)
+	if err != nil {
+		return diag.Errorf("error creating CES dashboard: ID is not found in API response")
+	}
+	d.SetId(id.(string))
+
+	rowWidgetNumNeedUpdate := false
+	dashboard, err := getDashboard(client, d)
+	if err != nil {
+		return diag.Errorf("error retrieving CES dashboard: %s", err)
+	}
+	rawSpecifiedRowWidgetNum := d.Get("row_widget_num")
+	rowWidgetNum := utils.PathSearch("row_widget_num", dashboard, float64(0)).(float64)
+	specifiedRowWidgetNum, _ := strconv.Atoi(rawSpecifiedRowWidgetNum.(string))
+	rowWidgetNumNeedUpdate = specifiedRowWidgetNum != int(rowWidgetNum)
+
+	_, isFavoriteOk := d.GetOk("is_favorite")
+	if rowWidgetNumNeedUpdate || isFavoriteOk {
+		err = updateDashboard(client, d)
+		if err != nil {
+			return diag.Errorf("error updating CES dashboard: %s", err)
+		}
+	}
+	return resourceDashboardRead(ctx, d, meta)
+}
+
+func buildCreateDashboardBodyParams(d *schema.ResourceData, cfg *config.Config) map[string]interface{} {
+	rawRowWidgetNum := d.Get("row_widget_num")
+	rowWidgetNum, _ := strconv.Atoi(rawRowWidgetNum.(string))
+	return map[string]interface{}{
+		"dashboard_name": d.Get("name"),
+		"row_widget_num": rowWidgetNum,
+		"dashboard_id":   utils.ValueIgnoreEmpty(d.Get("dashboard_id")),
+		"enterprise_id":  utils.ValueIgnoreEmpty(common.GetEnterpriseProjectID(d, cfg)),
```

Comment:
suggest to use the func in config
```suggestion
		"enterprise_id":  utils.ValueIgnoreEmpty(cfg.GetEnterpriseProjectID(d)),
```